### ISSUE TITLE
Wildcard resolving doesn't work on Android?

### ISSUE THREAD

**Issue by ropstah** [kapetan/dns#32]
I'm doing a very basic implementation of the DNS server:

`// Resolve these domain to localhost`
`server.MasterFile.AddIPAddressResourceRecord("test.local", "192.168.1.100");`
`server.MasterFile.AddIPAddressResourceRecord("*.extrasub.local", "192.168.1.100");`
`server.MasterFile.AddIPAddressResourceRecord("*.local", "192.168.1.100");`

On my Windows machine hosting the DNS server all three records are resolving correctly. However on my Android phone (setup via WiFi on the same network, using the specified DNS), only resolves `test.local` and doesn't work for the wildcards. 

Any ideas wether this is some DNS protocol thingy or something wrong with the library?

**Comment by ropstah** [comment]
Ok through some debugging I found a way to make it work. 

If I register an `AddIPAddressResourceRecord` like described it gets matched correctly by the `MasterFile.Matches` method, however a `ResponseException` is thrown during the `HandleRequest` methods `udp.SendAsync` call.

The Android client somehow doesn't like being returned the wildcard record directly (`NameError`). Strangely Windows does accept it.

Looking at how my domain provider DNS provided the question/answer results I was able to make it work. 

I basically intercepted the response and in the case it contains a wildcard `ResourceRecord` I remove this from the answers and create "virtual" `CanonicalNameResourceRecord` and fetch the "main" (non-wildcard) domain to which it should point as `IPAddressResourceRecord` and add both to the `response.Answers` list respectively. This seems to work.
